File: Day172/transposeMatrix.java

class Solution
{
    public void transpose(int n,int arr[][])
    {
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(i!=j && j<=i){
                    arr[i][j] = arr[i][j] ^ arr[j][i];
                    arr[j][i] = arr[j][i] ^ arr[i][j];
                    arr[i][j] = arr[i][j] ^ arr[j][i];
                }
            }
        }
    }
} 
    

File: Day325/TransposeMatrix.go
func transpose(matrix [][]int) [][]int {
    r, c := len(matrix), len(matrix[0])
    arr := make([][]int, c)
    for i:=0; i<c; i++ {
        arr[i] = make([]int, r)
        for j:=0; j<r; j++ {
            arr[i][j] = matrix[j][i]
        }
    }
    
    return arr
}